charmbits/service/server.go

```diff
@@ -49,29 +45,51 @@ type Context struct {
 	socketPath string
 }
 
+type rpcCommand struct {
+	listener net.Listener
+}
+
+func (c *rpcCommand) Kill() {
+	c.listener.Close()
+}
+
+func (c *rpcCommand) Wait() error {
+	return nil
```

Comment:
I've fixed it now to use a tomb properly, and added a test which was needed anyway.